lib/src/model/directives/hide_constant_implementations.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:dartdoc/src/model/documentation_comment.dart';
+import 'package:dartdoc/src/model/model_element.dart';
+
+final _hideConstantImplementationsRegExp =
+    RegExp(r'{@hideConstantImplementations}');
+
+/// Implement parsing the hideConstantImplementations dartdoc directive
+/// for this [ModelElement].  Used by [Container].
+mixin HideConstantImplementations on DocumentationComment {
+  bool? _hideConstantImplementations;
+
+  /// [true] if the {@hideConstantImplementations} dartdoc directive is present
+  /// in the documentation for this class.
+  bool get hideConstantImplementations {
+    if (_hideConstantImplementations == null) {
+      buildDocumentationAddition(documentationComment);
+    }
+    return _hideConstantImplementations!;
+  }
+
+  /// Hides [hideConstantImplementations] from doc while leaving a note to
+  /// ourselves to change rendering for these constants.
+  /// Example:
+  ///
+  ///     {@hideConstantImplementations}
+  @override
+  String buildDocumentationAddition(String rawDocs) {
```

Comment:
With this implementation, do we need to parse every doc comment _n_ times given _n_ directives? (Assuming we ask each element questions like `hideConstantImplementations` once each element)

Reply:
Let's benchmark this then to see that another parse of each doc comment doesn't slow down, e.g. flutter doc generation.

Reply:
We can also add this to an issue I have regarding duplicate parsing of comments: https://github.com/dart-lang/dartdoc/issues/2970

Reply:
flutter doc generation timing:

new:  2492.9 seconds
old: 2518.7 seconds

~1% difference, within margin of error (I don't think this code actually improves things, so this is definitely a margin-of-error situation).   As this was running I realized the performance impact isn't as bad as I first implied.   We are increasing the number of parse-throughs per documentation comment, but this only increases the amount of time for:

1) For Containers (Class, Mixin, Enum), only those that contain at least one constant will be impacted.
2) For every element type, only those that contain at least one documentation comment will have this impact (there's a shortcut taken via a regex).   This overlaps with Containers, of course.

This explains why the measurable impact is negligible here.   It's still a code smell, though, and we should fix this longer term.

Reply:
Yeah I'm not surprised. Good news. Thanks for checking!